renormalizer/mps/mp.py

```diff
@@ -166,16 +166,22 @@ def is_right_canon(self):
         return self.qnidx == 0
 
     def ensure_left_canon(self, rtol=1e-5, atol=1e-8):
-        if not self.check_left_canonical(rtol, atol):
+        if (not self.check_left_canonical(rtol, atol)) or self.to_right or\
```

Comment:
There is a minor optimization at this point: the conditions that do not need extensive calculation such as `self.to_right` could be moved before those that require extensive calculation such as `self.check_left_canonical`

Reply:
good point.